relayer/relayer.go

```diff
@@ -343,12 +348,32 @@ func (r *Relayer) RelayMessage(warpLogInfo *vmtypes.WarpLogInfo, metrics *Messag
 	r.currentRequestID++
 
 	// Update the database with the latest processed block height
-	err = r.db.Put(r.sourceBlockchainID, []byte(database.LatestProcessedBlockKey), []byte(strconv.FormatUint(warpLogInfo.BlockNumber, 10)))
-	if err != nil {
-		r.logger.Error(
-			fmt.Sprintf("failed to put %s into database", database.LatestProcessedBlockKey),
-			zap.Error(err),
-		)
+	if storeProcessedHeight {
+		// First, check that the stored height is less than the current block height
+		// This is necessary because the relayer may be processing blocks out of order on startup
+		latestProcessedBlockData, err := r.db.Get(r.sourceBlockchainID, []byte(database.LatestProcessedBlockKey))
```

Comment:
I'd recommend a helper for `getLatestProcessedBlock` also, that handles the DB read and conversion to a `big.Int`

Reply:
Added